### PR TITLE
espresso: add 6.1.0

### PR DESCRIPTION
**Pull request by davydden**
~~trying to get 6.1.0 to compile with `elpa`. PITA...~~

tested on Centos7 with gcc 4.8.5 and openmpi.

### CODE REVIEW

File: var/spack/repos/builtin/packages/elpa/package.py [hunk 1]
```diff
@@ -32,6 +32,7 @@ class Elpa(AutotoolsPackage):
     homepage = 'http://elpa.mpcdf.mpg.de/'
     url = 'http://elpa.mpcdf.mpg.de/elpa-2015.11.001.tar.gz'
 
+    version('2016.11.001.pre', '5656fd066cf0dcd071dbcaf20a639b37')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/elpa/package.py]: Is this a pre-release? We may want to mark the last stable release as `preferred`.

File: var/spack/repos/builtin/packages/elpa/package.py [hunk 2]
```diff
@@ -49,8 +50,18 @@ def url_for_version(self, version):
             t = 'http://elpa.mpcdf.mpg.de/elpa-{0}.tar.gz'
         return t.format(str(version))
 
-    def setup_environment(self, spack_env, run_env):
+    # override default implementation which returns static lib
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/elpa/package.py]: I think I actually have a fix for this. See:
https://github.com/LLNL/spack/pull/3367/files#r105244781
but I'm fine with making it explicit in this package until #3367 is merged.

File: var/spack/repos/builtin/packages/espresso/package.py [hunk 3]
```diff
@@ -84,7 +91,18 @@ def install(self, spec, prefix):
...
+            if spec.satisfies('@:5.4.0'):
+                options.append('--with-elpa=yes')
+            else:
+                options.extend([
+                    '--with-elpa-include={0}'.format(
+                        join_path(spec['elpa'].prefix,
+                                  'include',
+                                  'elpa-{0}'.format(str(spec['elpa'].version)),
+                                  'modules')
+                        ),
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/espresso/package.py]: `flake8` is complaining that:
```console
var/spack/repos/builtin/packages/espresso/package.py:103: [E123] closing bracket does not match indentation of opening bracket's line
```

File: var/spack/repos/builtin/packages/espresso/package.py [hunk 4]
```diff
@@ -69,7 +77,6 @@ def check_variants(self, spec):
             raise RuntimeError(error.format(variant='elpa'))
 
     def install(self, spec, prefix):
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/espresso/package.py]: It seems the maintainers of Quantum Espresso fixed `make install` at some point, and now we are installing in:
```console
<prefix>/bin/bin
```
Can you please also take care of that?

Comment (davydden) [inline var/spack/repos/builtin/packages/espresso/package.py]: done.

**Pull request merged**